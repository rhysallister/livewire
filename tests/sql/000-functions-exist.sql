\unset ECHO
\i tests/tests.sql

/*******************************************************************************
				Test Suite for LiveWire
*******************************************************************************/

SELECT plan(30);

SELECT has_extension('public', 'livewire', 'Check for the existense of the livewire extension');
SELECT has_function('public', 'lw_addedgeparticipant', ARRAY['text', 'json'], 'Check for lw_addedgeparticipant');
SELECT has_function('public', 'lw_addnodeparticipant', ARRAY['text', 'json'], 'Check for lw_addnodeparticipant');
SELECT has_function('public', 'lw_edgedelete', 'Check for lw_edgedelete');
SELECT has_function('public', 'lw_edgeinsert', 'Check for lw_edgeinsert');
SELECT has_function('public', 'lw_edgeupdate', 'Check for lw_edgeupdate');
SELECT has_function('public', 'lw_endnodes', ARRAY['text'], 'Check for lw_endnodes');
SELECT has_function('public', 'lw_generateedge', ARRAY['text', 'text'], 'Check for lw_generatedege');
SELECT has_function('public', 'lw_generatenode', ARRAY['text', 'text'], 'Check for lw_generatenode');
SELECT has_function('public', 'lw_generate', ARRAY['text'], 'Check for lw_generate');
SELECT has_function('public', 'lw_initialise', ARRAY['text', 'integer', 'double precision', 'boolean', 'text'], 'Check for lw_initialise');
SELECT has_function('public', 'lw_nodedelete', 'Check for lw_nodedelete');
SELECT has_function('public', 'lw_nodeinsert', 'Check for lw_nodeinsert');
SELECT has_function('public', 'lw_nodeupdate', 'Check for lw_nodeupdate');
SELECT has_function('public', 'lw_originrow', ARRAY['text', 'bigint'], 'Check for lw_originrow');
SELECT has_function('public', 'lw_redirect', ARRAY['text', 'bigint'], 'Check for lw_redirect');
SELECT has_function('public', 'lw_singlesource', ARRAY['text'], 'Check for lw_singlesource');
SELECT has_function('public', 'lw_singlesource', ARRAY['text', 'bigint'], 'Check for lw_singlesource - Single source variant');
SELECT has_function('public', 'lw_sourcenodes', ARRAY['text'], 'Check for lw_sourcenodes');
SELECT has_function('public', 'lw_srid', ARRAY['text'], 'Check for lw_srid');
SELECT has_function('public', 'lw_tableconfig', ARRAY['text','text'], 'Check for lw_tableconfig');
SELECT has_function('public', 'lw_tolerance', ARRAY['text'], 'Check for lw_tolerance');
SELECT has_function('public', 'lw_traceall', ARRAY['text'], 'Check for lw_traceall');
SELECT has_function('public', 'lw_tracednstream', ARRAY['text','bigint'], 'Check for lw_tracednstream - Single source variant');
SELECT has_function('public', 'lw_tracednstream', ARRAY['text','bigint[]'], 'Check for lw_tracednstream - Multiple source variant');
SELECT has_function('public', 'lw_tracednstream', ARRAY['text','bigint[]','bigint[]'], 'Check for lw_tracednstream - Multiple source and blockers variant');
SELECT has_function('public', 'lw_tracednstream', ARRAY['text','geometry'], 'Check for lw_tracednstream - Multiple source and blockers variant');
SELECT has_function('public', 'lw_tracesource', ARRAY['text', 'bigint','boolean'], 'Check for lw_tracesource');
SELECT has_function('public', 'lw_traceupstream', ARRAY['text','bigint'], 'Check for lw_traceupstream');
SELECT has_function('public', 'lw_version', 'Check for lw_version');
SELECT * FROM finish();

ROLLBACK;
