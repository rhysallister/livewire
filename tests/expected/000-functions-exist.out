\unset ECHO
1..30
ok 1 - Check for the existense of the livewire extension
ok 2 - Check for lw_addedgeparticipant
ok 3 - Check for lw_addnodeparticipant
ok 4 - Check for lw_edgedelete
ok 5 - Check for lw_edgeinsert
ok 6 - Check for lw_edgeupdate
ok 7 - Check for lw_endnodes
ok 8 - Check for lw_generatedege
ok 9 - Check for lw_generatenode
ok 10 - Check for lw_generate
ok 11 - Check for lw_initialise
ok 12 - Check for lw_nodedelete
ok 13 - Check for lw_nodeinsert
ok 14 - Check for lw_nodeupdate
ok 15 - Check for lw_originrow
ok 16 - Check for lw_redirect
ok 17 - Check for lw_singlesource
ok 18 - Check for lw_singlesource - Single source variant
ok 19 - Check for lw_sourcenodes
ok 20 - Check for lw_srid
ok 21 - Check for lw_tableconfig
ok 22 - Check for lw_tolerance
ok 23 - Check for lw_traceall
ok 24 - Check for lw_tracednstream - Single source variant
ok 25 - Check for lw_tracednstream - Multiple source variant
ok 26 - Check for lw_tracednstream - Multiple source and blockers variant
ok 27 - Check for lw_tracednstream - Multiple source and blockers variant
ok 28 - Check for lw_tracesource
ok 29 - Check for lw_traceupstream
ok 30 - Check for lw_version
